Source/NSubstitute/Core/CallResults.cs

```diff
@@ -6,12 +6,18 @@ namespace NSubstitute.Core
     public class CallResults : ICallResults
     {
         readonly ICallInfoFactory _callInfoFactory;
+        private readonly bool _skipVoidMethods;
         ConcurrentQueue<ResultForCallSpec> _results;
 
-        public CallResults(ICallInfoFactory callInfoFactory)
+        public CallResults(ICallInfoFactory callInfoFactory, bool skipVoidMethodsForPerformance)
```

Comment:
No problem, let me show you the very simplified example of the AutoFixture custom handler implementation (how I imagine it). [See it here](https://gist.github.com/Zvirja/e58f5390bf025b6fcce1d6684fcac3f8). 

It uses __own instance__ of `CallResults` to store already resolved values. I need `CallResults` that doesn't skip void methods, so I want just to derive from `CallResults`, override that method and disable "skip void" behavior.

Let me know if you see the purpose now. If not - clarify what exactly is confusing you :)